app/src/main/java/com/grofers/hikingTrailsMap/maphiking/MapsActivity.java

```diff
@@ -0,0 +1,477 @@
+        String url = "http://192.168.1.55:5000/trails/create";
+        String jsonString = gson.toJson(trailDetail);
+        JSONObject jsonObject = null;
+        try {
+            jsonObject = new JSONObject(jsonString);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        PostJsonData postJsonData = new PostJsonData(Request.Method.POST, url, jsonObject, new Response.Listener<JSONObject>() {
+            @Override
+            public void onResponse(JSONObject response) {
+                Log.d("Response: ", response.toString());
+                progressDialog.hide();
+                shareTrail();
+                trail.clear();
+                mMap.clear();
+            }
+        }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError response) {
+                Log.d("Response: ", response.toString());
+                progressDialog.hide();
+                //Todo: share trail must not be here
+                shareTrail();
+                mMap.clear();
+
+            }
+        });
+
+        postJsonData.setRetryPolicy(new DefaultRetryPolicy(
+                5000,
+                DefaultRetryPolicy.DEFAULT_MAX_RETRIES,
+                DefaultRetryPolicy.DEFAULT_BACKOFF_MULT));
+        queue.add(postJsonData);
+
+    }
+
+    private void shareTrail()
+    {
+        //String shareUrl = "https://www.google.com/maps?saddr=San+Francisco&daddr=GooglePlex+Mountain+View+" +
+        //        "to:Google+Building+45+to:San+Jose";
+        String shareUrl = "https://www.google.com/maps?f=d&hl=en";
+        String params = "";
+        int size = trail.size();
+        Log.d("trail length", String.valueOf(size));
+        params = params + "saddr=" + trail.get(0).latitude + "," + trail.get(0).longitude;
+        params = params + "&";
+        params = params + "daddr=" + trail.get(size - 1).latitude + "," + trail.get(size - 1).longitude;
+        for(int i = 1; i < size - 1; i++)
+        {
+            params = params + "+to:" + trail.get(i).latitude + "," + trail.get(i).longitude;
+        }
+        //String urlFormat = "http://maps.google.com/maps?saddr=20.344,34.34&daddr=20.5666,45.345";
+        shareUrl = shareUrl + params;
+        Intent shareIntent  = new Intent(Intent.ACTION_SEND);
+        shareIntent.setType("text/plain");
+        shareIntent.putExtra(Intent.EXTRA_SUBJECT, "My New Trail");
+        shareIntent.putExtra(Intent.EXTRA_TEXT, shareUrl);
+        startActivity(shareIntent);
+    }
+
+    private void searchTrail(Double lat, Double lon)
+    {
+        final ProgressDialog progressDialog;
+        progressDialog = new ProgressDialog(context);
+        progressDialog.setMessage("Getting Nearby Trails, Please wait...");
+        progressDialog.setIndeterminate(true);
+        progressDialog.show();
+
+        String url = "http://192.168.1.55:5000/trails/search?lat=" + lat + "&" + lon;
```

Comment:
use StringBuffer for better practices. No such performance cost in this case but in case of loop concatenation, StringBuffers are much more efficient than + operator.